src/v/cluster/rm_partition_frontend.cc

```diff
@@ -312,8 +312,10 @@ ss::future<begin_tx_reply> rm_partition_frontend::do_begin_tx(
                     }
                     return begin_tx_reply{ntp, tx_errc::unknown_server_error};
                 }
-
-                return begin_tx_reply{ntp, etag.value(), tx_errc::none};
+                auto topic_md = _metadata_cache.local().get_topic_metadata(
+                  model::topic_namespace_view(ntp));
+                return begin_tx_reply{
+                  ntp, etag.value(), tx_errc::none, topic_md->get_revision()};
```

Comment:
what guarantees that `get_topic_metadata` doesn't return `std::nullopt` leading to undefined behavior when dereference the optional?